test/test_script.py

```diff
@@ -180,6 +180,15 @@ def test_cmd_lower(self):
     def test_cmd_upper(self):
         self.assertScriptResultEquals("$upper(AbeCeDA)", "ABECEDA")
 
+    def test_cmd_pad(self):
+        self.assertScriptResultEquals("$pad(abc de,10,-)", "----abc de")
+
+    def test_cmd_replace(self):
+        self.assertScriptResultEquals("$replace(abc ab abd a,ab,test)", "testc test testd a")
+
+    def test_cmd_strip(self):
+        self.assertScriptResultEquals("$strip(   abc  de f  )", "abc de f")
```

Comment:
For strip we should make sure it removes other whitespace characters then pnly space. I would additionally test for newlines `\n` and tabs `\t` by modifying the test to include those characters, e.g. `"$strip(  \t abc  de f \n )", "abc de f"`